libuavcan_drivers/linux/include/uavcan_linux/socketcan.hpp

```diff
@@ -225,6 +227,16 @@ class SocketCanIface : public uavcan::ICanIface
         {
             return (res < 0 && errno == EWOULDBLOCK) ? 0 : res;
         }
+        /*
+         * Flags
+         */
+        loopback = (msg.msg_flags & static_cast<int>(MSG_CONFIRM)) != 0;
+
+        if (!checkHWFilters(sockcan_frame) && !loopback)
```

Comment:
Order should be reversed: `if (!loopback && !checkHWFilters(sockcan_frame))`, because there is no point in checking HW filters on loopback frames.